src/prefect/tasks/airbyte/airbyte.py

```diff
@@ -0,0 +1,296 @@
+                self.logger.info(response.json())
+                job_status = response.json()["job"]["status"]
+                job_created_at = response.json()["job"]["createdAt"]
+                job_updated_at = response.json()["job"]["updatedAt"]
+                return job_status, job_created_at, job_updated_at
+            elif response.status_code == 404:
+                self.logger.error(f"Job {job_id} not found...")
+                raise JobNotFoundException(f"Job {job_id} not found...")
+        except RequestException as e:
+            raise AirbyteServerNotHealthyException(e)
+
+    @defaults_from_attrs(
+        "airbyte_server_host",
+        "airbyte_server_port",
+        "airbyte_api_version",
+        "connection_id",
+    )
+    def run(
+        self,
+        airbyte_server_host: str,
+        airbyte_server_port: int,
+        airbyte_api_version: str,
+        connection_id: str,
+        poll_interval_s: int = 15,
+    ) -> dict:
+        """
+        Task run method for triggering an Airbyte Connection.
+
+        *It is assumed that the user will have previously configured
+        a Source & Destination into a Connection.*
+        e.g. MySql -> CSV
+
+        An invocation of `run` will attempt to start a sync job for
+        the specified `connection_id` representing the Connection in
+        Airbyte.
+
+        `run` will poll Airbyte Server for the Connection status and
+        will only complete when the sync has completed or
+        when it receives an error status code from an API call.
+
+        Args:
+            - airbyte_server_host (str, optional): Hostname of Airbyte server where connection is
+                configured. Will overwrite the value provided at init if provided.
+            - airbyte_server_port (str, optional): Port that the Airbyte server is listening on.
+                Will overwrite the value provided at init if provided.
+            - airbyte_api_version (str, optional): Version of Airbyte API to use to trigger connection
+                sync. Will overwrite the value provided at init if provided.
+            - connection_id (str, optional): if provided,
+                will overwrite the value provided at init.
+            - poll_interval_s (int, optional): this task polls the
+                Airbyte API for status, if provided this value will
+                override the default polling time of 15 seconds.
+
+        Returns:
+            - dict: connection_id (str) and succeeded_at (timestamp str)
+        """
+        if not connection_id:
+            raise ValueError(
+                "Value for parameter `connection_id` *must* \
+            be provided."
+            )
+
+        uuid = re.compile("^[0-9A-Fa-f-]+$")
+        match = uuid.match(connection_id)
+        if not match:
+            raise ValueError(
+                "Parameter `connection_id` *must* be a valid UUID \
+                i.e. 32 hex characters, including hyphens."
+            )
```

Comment:
I think you can do

```
import uuid

try:
    uuid.UUID(connection_id)
except ValueError as exc:
    raise ...
```

which I think covers this check better? Not sure what the valid ids are though. This could also take `connection_id` as a `UUID` type directly? Or `Union[UUID, str]`